em_examples/HarmonicVMDCylWidget.py

```diff
@@ -181,139 +246,193 @@ def getData(self, bType="b"):
-                tname = "Im("
-            elif ComplexNumber == "amplitude":
-                tname = "Amp("
-            elif ComplexNumber == "phase":
-                tname = "Phase("
-            title = tname + Field + view+")-field"
-
-        if Field == "B":
-            label = "Magnetic field (T)"
+            fig = plt.figure(figsize=(10, 6))
+            ax = plt.subplot(111)
+            vec = False
             if view == "vec":
-                vec = True
+                tname = "Vector "
+                title = tname+Field+"-field"
+            elif view == "amp":
+                tname = "|"
+                title = tname+Field+"|-field"
+            else:
                 if ComplexNumber == "real":
-                    val = np.c_[self.Bx.real, self.Bz.real]
+                    tname = "Re("
                 elif ComplexNumber == "imag":
-                    val = np.c_[self.Bx.imag, self.Bz.imag]
+                    tname = "Im("
+                elif ComplexNumber == "amplitude":
+                    tname = "Amp("
+                elif ComplexNumber == "phase":
+                    tname = "Phase("
+                title = tname + Field + view+")-field"
+
+            if Field == "B":
+                label = "Magnetic field (T)"
+                if view == "vec":
+                    vec = True
+                    if ComplexNumber == "real":
+                        val = np.c_[self.Bx.real, self.Bz.real]
+                    elif ComplexNumber == "imag":
+                        val = np.c_[self.Bx.imag, self.Bz.imag]
+                    else:
+                        # ax.imshow(self.im)
+                        ax.set_xticks([])
+                        ax.set_yticks([])
+                        return "Vector plot only supports real and imaginary type!"
+
+                elif view == "x":
+                    if ComplexNumber == "real":
+                        val = self.Bx.real
+                    elif ComplexNumber == "imag":
+                        val = self.Bx.imag
+                    elif ComplexNumber == "amplitude":
+                        val = abs(self.Bx)
+                    elif ComplexNumber == "phase":
+                        val = np.angle(self.Bx)
+                elif view == "z":
+                    if ComplexNumber == "real":
+                        val = self.Bz.real
+                    elif ComplexNumber == "imag":
+                        val = self.Bz.imag
+                    elif ComplexNumber == "amplitude":
+                        val = abs(self.Bz)
+                    elif ComplexNumber == "phase":
+                        val = np.angle(self.Bz)
                 else:
                     # ax.imshow(self.im)
                     ax.set_xticks([])
                     ax.set_yticks([])
-                    return "Vector plot only supports real and imaginary type!"
+                    print("Think about the problem geometry. There is NO By in this case.")
 
```

Comment:
it is also worth adding a `return` statement right after this (no need to draw the figure, printing is enough)